### ISSUE TITLE
Release v6.44.0 removed support for `cf oauth-token` with client credentials

### ISSUE THREAD

**Issue by matt-royal** [cloudfoundry/cli#1644]
**Please fill out the issue checklist below and provide ALL the requested information.**

- [x] I tried updating to the latest version of the CF CLI to see if it fixed my problem.
- [x] I attempted to run the command with `CF_TRACE=1` to help debug the issue.
- [x] I am reporting a bug that others will be able to reproduce.

**Describe the bug and the command you saw an issue with**
My team's deployment tooling relies on the `cf oauth-token` command. We also authenticate as a client using `cf auth --client-credentials`. As documented in the 6.44.0 release, this combination no longer works. 

**What happened**
We receive a `Credentials were rejected, please try again` error message when calling `cf oauth-token` while authed as a client.

**Expected behavior**
The command prints a valid oauth token without erroring.

**To Reproduce**
Steps to reproduce the behavior; include the exact CLI commands and verbose output:
1. Run `cf auth --client-credentials $CLIENT_ID $CLIENT_SECRET`
1. Run `cf oauth-token`
1. Observe the `Credentials were rejected, please try again` error

**Provide more context**
This has been reproduced on Mac OS X 10.14.3, running CLI 6.44.1+c3b20bfbe.2019-05-08
The CC API version was 2.134.0 (on PWS)

My team is using `cf oauth-token` to implement our own version of `cf curl` that supports the `-f` flag. As you've recently added support for that flag to the built-in `cf curl` command, we could switch to that, but we'd prefer not to have to patch the past few versions of our product for this backwards-incompatible change.

Note: As of January 2019, we no longer support API versions older than CF Release v284/CF Deployment v1.7.0 (CAPI Release: 1.46.0 (APIs 2.100.0 and 3.35.0).

**Comment by acrmp** [comment]
Hi Matt,

Sorry we broke your workflow. We intend to fix this but I want to share the current behaviour and see if there is any feedback on our proposed approach.

The reason it broke is because of a change in 6.44.0 around how client credentials were handled, and how this interacts with the existing behaviour of `cf oauth-token`.

* When authenticating with client credentials version 6.44.0 of the cli includes a change to remove the client secret from the config file.
* The current behaviour of the `cf oauth-token` command is not (as you might expect) to return the current token, but to always refresh the token and output the refreshed token.
* UAA clients do not have a refresh token. Prior to 6.44.0 the cli would re-authenticate using the client id and secret from the config file. However as the secret has been removed the authentication will now fail.

**cf-oauth-token behaviour prior to 6.44.0**

token status | client credentials | username & password
-- | -- | --
current | re-authenticate and return | refresh and return
expired | re-authenticate and return | refresh and return
no token (not logged in) | fail | fail

**cf-oauth-token current behaviour (6.44.1)**

token status | client credentials | username & password
-- | -- | --
current | fail | refresh and return
expired | fail | refresh and return
no token (not logged in) | fail | fail

**cf-oauth-token proposed behaviour**

* It's not possible to re-authenticate and return a new token when using client credentials. We no longer have access to the client secret to do this.
* We can return the current token when using client credentials. We have the option here of either returning the token always (including when it has expired, causing a subsequent call using the token to fail) or checking the token validity and failing when the token is invalid. We're leaning towards the later.
* For users we can either keep the existing behaviour and always refresh the token, or return the existing token without refreshing. We're leaning towards always refreshing the token so that we are reducing the risk of breakage in case someone relies on the current side-effect of the token being refreshed.

token status | client credentials | username & password
-- | -- | --
current | return current token | refresh and return
expired | fail | refresh and return
no token (not logged in) | fail | fail

While we hope to implement this soon, we'd love to hear any feedback on the proposed approach above.

Thanks,

Andrew.
